src/BreakpointViewer.cpp

```diff
@@ -856,22 +851,24 @@ std::optional<Breakpoint> BreakpointViewer::parseTableRow(BreakpointRef::Type ty
 	return bp;
 }
 
-
 void BreakpointViewer::onSymbolTableChanged()
 {
 	for (int row = 0; row < bpTableWidget->rowCount(); ++row) {
-		auto* item = bpTableWidget->item(row, BP_ADDRESS);
+		auto* item = bpTableWidget->item(row, LOCATION);
 
-		// scan tooltip validity
-		if (!item->text().isEmpty()) {
-			Symbol* s = debugSession.symbolTable().getAddressSymbol(item->text());
-			auto address = stringToValue<uint16_t>(item->text());
-			assert(address);
+		if (auto* ref = scanBreakpointRef(BreakpointRef::BREAKPOINT, row)) {
```

Comment:
This implementation makes sense, I think it's correct. However it's different from the 'recover' functionality I see in e.g. `parseSymbolOrValue()` (maybe also in `parseSlotField()` and `parseSegmentField()`).

More specifically I'm talking about the operations to go from a row-number to the Breakpoint object:
* `parseSymbolOrValue()` does (simplified) `breakpoints->getBreakpoint(findBreakpointRef(row)->index)`
* `onSymbolTableChanged()` does (simplified) `breakpoints->getBreakpoint(scanBreakpointRef(row)->index)`

So `findBreakpointRef()` vs `scanBreakpointRef()`. I'm not sure what the difference is, but I would expect we need the same function twice.

Reply:
This one is really my only blocking question to accept this PR. (And maybe the answer is that the current code is correct: it really should be `findBreakpointRef()` and `scanBreakpointRef()`.)